cpp/src/aztec/lagrange_base_gen/lagrange_base_processor.cpp

```diff
@@ -44,10 +44,35 @@ int main(int argc, char** argv)
     g2_elements.push_back(verifier_ref_string->get_g2x());
     g2_elements.push_back(barretenberg::g2::affine_one);
 
-    barretenberg::io::Manifest manifest{
-        0, 1, static_cast<uint32_t>(subgroup_size), 2, static_cast<uint32_t>(subgroup_size), 2, 0
-    };
-    barretenberg::io::write_transcript(&lagrange_base_srs[0], &g2_elements[0], manifest, lagrange_srs_path, true);
+    const uint32_t subgroup_size_ = static_cast<uint32_t>(subgroup_size);
+    const uint32_t max_lagrange_transcript_size = static_cast<uint32_t>((size_t(1) << 24));
+    if (subgroup_size > max_lagrange_transcript_size) {
```

Comment:
Do we really need the if-else here? Seems like the code within the if could handle both cases (with some minor modifications perhaps). Maybe I'm missing a subtlety. Not a big deal either way

Reply:
Good point, simplified it. Thank you!